compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BFiniteType.java

```diff
@@ -65,6 +65,41 @@ public String toString() {
         return joiner.toString();
     }
 
+    @Override
+    public boolean hasImplicitInitialValue() {
+        // Has NIL element as a member.
+        if (valueSpace.stream().anyMatch(value -> value.type.tag == TypeTags.NIL)) {
+            return true;
+        }
+        // All of the element are from same type. And elements contains implicit initial value.
+        if (valueSpace.isEmpty()) {
+            return false;
+        }
+
+        BLangExpression firstElement = valueSpace.iterator().next();
+        boolean sameType = valueSpace.stream().allMatch(value -> value.type.tag == firstElement.type.tag);
+        if (!sameType) {
+            return false;
+        }
+        switch (firstElement.type.tag) {
+            case TypeTags.STRING:
+                return containsElement(valueSpace, "\"\"");
+            case TypeTags.INT:
+                return containsElement(valueSpace, "0");
+            case TypeTags.FLOAT:
+            case TypeTags.DECIMAL:
+                return containsElement(valueSpace, "0.0");
```

Comment:
Please check, is this working in a scenario like this, 
```
 type Foo <decimal> 0 | 10;
```

Reply:
At the moment we don't support this syntax, will add check for decimal "0" for future proofing.